src/webextension/background/views.js

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+const views = {
+  "firstrun": null,
+  "manage": [],
+};
+
+(function setupFirstRun(views) {
+  let tabId;
+
+  views.firstrun = {
+    open: async() => {
+      let windowId;
+      if (tabId !== undefined) {
+        // verify tab still exists
+        try {
+          let tabInfo = await browser.tabs.get(tabId);
+          windowId = tabInfo.windowId;
+        } catch (err) {
+          // does not exist, forget existing info
+          tabId = undefined;
+        }
+      }
+
+      if (tabId !== undefined) {
+        // focus owning window and activate tab
+        await browser.windows.update(windowId, {
+          focused: true,
+        });
+        await browser.tabs.update(tabId, {
+          active: true,
+        });
+      } else {
+        // create a tab in the current window
+        let tabInfo = await browser.tabs.create({
+          url: browser.extension.getURL("/firstrun/index.html"),
+        });
+        tabId = tabInfo.id;
+      }
+      return tabId;
+    },
+    close: async() => {
+      try {
+        tabId && await browser.tabs.remove(tabId);
```

Comment:
I notice that you were careful above to check `tabId === undefined`; is this line safe, or could we run into issues where `tabId` is 0?

Reply:
that was a carry over from earlier.  The docs don't say `tabId` cannot be 0.

I'll change it to be more careful